contrib/PFAS_concentrations/PFAS_concentrations.py

```diff
@@ -0,0 +1,740 @@
+                "pkey_filter",
+                "aquifer_code",
+                "diepte_onderkant_filter",
+                "lengte_filter"])
+
+            data["datum_monstername"] = pd.to_datetime(
+                data["datum_monstername"])
+            data = pd.merge(data, filter_elements)
+        except ValueError as e:
+            logger.info(f"Empty dataframe: {e}")
+        return data
+
+    def biota(self, location, max_features, query=None, sort_by=None):
+        """
+        Download the biota data.
+
+        Parameters
+        ----------
+        location:
+            Query on location.
+            (https://pydov.readthedocs.io/en/stable/query_location.html)
+        max_features: int
+            Limit the number of WFS features you want to be returned.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+        query:
+            Find data based on one or more of its attribute values.
+            (https://pydov.readthedocs.io/en/stable/query_attribute.html)
+        sort_by:
+            Sort on one or multiple attributes.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+
+        Returns
+        -------
+        The downloaded biota data.
+        """
+        logger.info(f"Downloading biota data")
+        data_wfs_VMM_biota = self.wfs_request(
+            'pfas:pfas_biota',
+            location, max_features, query, sort_by)
+
+        data_wfs_VMM_biota = data_wfs_VMM_biota.drop_duplicates(
+            subset=data_wfs_VMM_biota.columns)
+
+        data_wfs_VMM_biota_len = len(data_wfs_VMM_biota)
+
+        nb_datapoints = {"Biota_VMM" : data_wfs_VMM_biota_len}
+        self.dictionary["nb_datapoints"][0].update(nb_datapoints)
+
+        return data_wfs_VMM_biota
+
+    def effluent(self, location, max_features, query=None, sort_by=None):
+        """
+        Download the effluent data.
+
+        Parameters
+        ----------
+        location:
+            Query on location.
+            (https://pydov.readthedocs.io/en/stable/query_location.html)
+        max_features: int
+            Limit the number of WFS features you want to be returned.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+        query:
+            Find data based on one or more of its attribute values.
+            (https://pydov.readthedocs.io/en/stable/query_attribute.html)
+        sort_by:
+            Sort on one or multiple attributes.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+
+        Returns
+        -------
+        The downloaded effluent data.
+        """
+        logger.info(f"Downloading effluent data")
+        data_wfs_OVAM = self.wfs_request(
+            'pfas:pfas_analyseresultaten',
+            location, max_features, query, sort_by)
+
+        data_wfs_OVAM = data_wfs_OVAM.drop_duplicates(
+            subset=data_wfs_OVAM.columns)
+
+        data_wfs_OVAM = data_wfs_OVAM[data_wfs_OVAM['medium'] == 'Effluent']
```

Comment:
This is very inefficient, I think this could be replaced with a `PropertyIsEqualTo` query in the `wfs_request`?

Currently, if you request `medium = ['all']` all data is downloaded 8 times, only to return it once.

Reply:
I thought this was less of an issue since the request is stored in the buffer, but I can make the modification.